infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -385,36 +385,43 @@ private boolean runRecoveryReconfiguration() {
      * These metrics are then composed into ServerMetrics model and stored locally.
      */
     private void updateLocalMetrics() {
-        // Initializing the status of components
-        // No need to poll unless node is bootstrapped.
-        Layout layout = serverContext.copyManagementLayout();
-        if (layout == null) {
-            return;
-        }
 
-        // Build and replace existing locally stored nodeMetrics
-        SequencerMetrics sequencerMetrics;
-        // This is an optimization. If this node is not the primary sequencer for the current
-        // layout, there is no reason to request metrics from this sequencer.
-        if (layout.getSequencers().get(0).equals(getLocalEndpoint())) {
-            try {
-                sequencerMetrics = CFUtils.getUninterruptibly(
-                        getCorfuRuntime()
-                                .getLayoutView().getRuntimeLayout(layout)
-                                .getSequencerClient(getLocalEndpoint())
-                                .requestMetrics());
-            } catch (ServerNotReadyException snre) {
-                sequencerMetrics = new SequencerMetrics(SequencerStatus.NOT_READY);
-            } catch (Exception e) {
-                log.error("Error while requesting metrics from the sequencer: ", e);
+        try {
+            // Initializing the status of components
+            // No need to poll unless node is bootstrapped.
+            Layout layout = serverContext.copyManagementLayout();
+            if (layout == null) {
+                return;
+            }
+
+            // Build and replace existing locally stored nodeMetrics
+            SequencerMetrics sequencerMetrics;
+            // This is an optimization. If this node is not the primary sequencer for the current
+            // layout, there is no reason to request metrics from this sequencer.
+            if (layout.getSequencers().get(0).equals(getLocalEndpoint())) {
+                try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)

Reply:
maybe we can refactor according to the automated suggestion in this case.

Reply:
Done.